src/main/java/com/example/demo/security/configuration/ApplicationSecurity.java

```diff
@@ -0,0 +1,18 @@
+package com.example.demo.security.configuration;
+
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.config.http.SessionCreationPolicy;
+
+@EnableWebSecurity
+public class ApplicationSecurity extends WebSecurityConfigurerAdapter {
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http.csrf().disable();
+        http.headers().frameOptions().disable();
+        http.authorizeRequests().anyRequest().permitAll();
```

Comment:
Does the `permitAll()` at the end of that line, overwrite the `authorizeRequests().anyRequest()`. Sorry this is a little confusing to me because the first two methods sound like we're going to authorise everything and then the last sounds like we're going to allow everyone in.

Reply:
I mostly copied these blindly, so I'd better look them properly!
- `http.csrf().disable();` disables protection against Cross Site Request Forgery. This protection is useful when the client is a browser. So in hindsight I should enable it! https://stackoverflow.com/questions/52363487/what-is-the-reason-to-disable-csrf-in-spring-boot-web-application
- `http.headers().frameOptions().disable();` disables protection against the browser rendering a page in a <frame> or <iframe>. The protection is there to prevent against clickjacking attacks. But I needed to disable it so that I can look at the h2-console :) https://stackoverflow.com/questions/65894268/how-does-headers-frameoptions-disable-work
- Finally `http.authorizeRequests().anyRequest().permitAll();` allows all http requests to go through spring security. In a future PR I will only permit requests to /auth/login, unless the request inclides the JWT